source/UF/Pushouts.lagda

```diff
@@ -0,0 +1,418 @@
+            (dependent-cocone-commuting-square f g pushout pushout-cocone P t)
+  ∼ ∼-trans (λ - → apd (pushout-unique-map t z) (glue -))
+            ((pushout-inrr-homotopy t z) ∘ g)
+ pushout-glue-homotopy t (u , K , L , M) = M
+
+\end{code}
+
+Now we can derive the induction and recursion principles along with the corresponding
+computation rules, the uniqueness of maps out of the pushout and the (non-dependent)
+universal property.
+
+\begin{code}
+
+ pushout-induction
+  : {P : pushout → 𝓣  ̇}
+  → Pushout-Induction-Principle pushout f g (inll , inrr , glue) P
+ pushout-induction l r G
+  = pushout-unique-map (l , r , G) (pushout-fiber-center (l , r , G))
+
+ pushout-ind-comp-inll
+  : {P : pushout → 𝓣  ̇}
+  → Pushout-Computation-Rule₁ pushout f g (inll , inrr , glue) P pushout-induction
+ pushout-ind-comp-inll l r G
+  = pushout-inll-homotopy (l , r , G) (pushout-fiber-center (l , r , G))
+
+ pushout-ind-comp-inrr
+  : {P : pushout → 𝓣  ̇}
+  → Pushout-Computation-Rule₂ pushout f g (inll , inrr , glue) P pushout-induction
+ pushout-ind-comp-inrr l r G
+  = pushout-inrr-homotopy (l , r , G) (pushout-fiber-center (l , r , G))
+
+ pushout-ind-comp-glue
+  : {P : pushout → 𝓣  ̇}
+  → Pushout-Computation-Rule₃ pushout f g (inll , inrr , glue) P
+     pushout-induction pushout-ind-comp-inll pushout-ind-comp-inrr
+ pushout-ind-comp-glue l r G c
+  = pushout-glue-homotopy (l , r , G) (pushout-fiber-center (l , r , G)) c ⁻¹
+   
+ pushout-recursion : {D : 𝓣  ̇}
+                   → (l : A → D)
+                   → (r : B → D)
+                   → ((c : C) → l (f c) ＝ r (g c))
+                   → pushout → D
+ pushout-recursion l r G =
+  pushout-induction l r (λ c → (transport-const (glue c) ∙ G c))
+
+ pushout-rec-comp-inll : {D : 𝓣  ̇}
+                       → (l : A → D)
+                       → (r : B → D)
+                       → (G : (c : C) → l (f c) ＝ r (g c))
+                       → (a : A)
+                       → pushout-recursion l r G (inll a) ＝ l a
+ pushout-rec-comp-inll l r G =
+  pushout-ind-comp-inll l r (λ c → (transport-const (glue c) ∙ G c))
+
+ pushout-rec-comp-inrr : {D : 𝓣  ̇}
+                       → (l : A → D)
+                       → (r : B → D)
+                       → (G : (c : C) → l (f c) ＝ r (g c))
+                       → (b : B)
+                       → pushout-recursion l r G (inrr b) ＝ r b
+ pushout-rec-comp-inrr l r G =
+  pushout-ind-comp-inrr l r (λ c → (transport-const (glue c) ∙ G c))
+
+ pushout-rec-comp-glue
+  : {D : 𝓣  ̇}
+  → (l : A → D)
+  → (r : B → D)
+  → (G : (c : C) → l (f c) ＝ r (g c))
+  → (c : C)
+  → ap (pushout-recursion l r G) (glue c) ∙ pushout-rec-comp-inrr l r G (g c) 
+  ＝ pushout-rec-comp-inll l r G (f c) ∙ G c
+ pushout-rec-comp-glue {𝓣} {D} l r G c =
+  ap (pushout-recursion l r G) (glue c) ∙ pushout-rec-comp-inrr l r G (g c)                                                                 ＝⟨ III ⟩
```

Comment:
The "III" here is far off with lots of spaces before it. You should remove these spaces.

Reply:
I think the problem may be that you thought that there is a line break but it is just emacs wrapping around a long line with lots of spaces.